DD4hep not working with older ROOT version

When using a root version prior to the introduction of TGeoSystemOfUnits the compilation fails
```
Building CXX object DDCore/CMakeFiles/DDCore.dir/src/DetectorImp.cpp.o
/cvmfs/clicdp.cern.ch/iLCSoft/builds/nightly/x86_64-slc6-gcc62-opt/DD4hep/HEAD/DDCore/src/DetectorImp.cpp:35:31: fatal error: TGeoSystemOfUnits.h: No such file or directory
 #include "TGeoSystemOfUnits.h"
```

--- a/DDCore/src/DetectorImp.cpp
+++ b/DDCore/src/DetectorImp.cpp
@@ -32,7 +32,9 @@
 #include <mutex>
 
 // ROOT inlcude files
+#if ROOT_VERSION_CODE >= ROOT_VERSION(6,12,0)
 #include "TGeoSystemOfUnits.h"
+#endif
 #include "TGeoCompositeShape.h"
 #include "TGeoBoolNode.h"
 #include "TGeoManager.h"
@@ -155,7 +157,9 @@ DetectorImp::DetectorImp()
 DetectorImp::DetectorImp(const string& name)
   : TNamed(), DetectorData(), DetectorLoad(this), m_buildType(BUILD_NONE)
 {
-  //TGeoUnit::setUnitType(TGeoUnit::kTGeant4Units);
+#if ROOT_VERSION_CODE >= ROOT_VERSION(6,20,0)
+  TGeoUnit::setUnitType(TGeoUnit::kTGeant4Units);
+#endif
   SetTitle("DD4hep detector description object");
   set_unexpected( description_unexpected );
   set_terminate( description_unexpected );

